air/src/execution_step/air/call/prev_result_handler.rs

```diff
@@ -71,24 +68,32 @@ pub(super) fn handle_prev_state<'i>(
                 // result hasn't been prepared yet
                 None => {
                     exec_ctx.subgraph_complete = false;
-                    Ok(StateDescriptor::not_ready(prev_result))
+                    Ok(StateDescriptor::not_ready(met_result.result))
                 }
             }
         }
         RequestSentBy(..) => {
             // check whether current node can execute this call
             let is_current_peer = tetraplet.peer_pk.as_str() == exec_ctx.run_parameters.current_peer_id.as_str();
             if is_current_peer {
-                return Ok(StateDescriptor::can_execute_now(prev_result));
+                return Ok(StateDescriptor::can_execute_now(met_result.result));
             }
 
             exec_ctx.subgraph_complete = false;
-            Ok(StateDescriptor::cant_execute_now(prev_result))
+            Ok(StateDescriptor::cant_execute_now(met_result.result))
         }
         // this instruction's been already executed
-        Executed(ref mut value) => {
-            set_result_from_value(&mut *value, tetraplet.clone(), trace_pos, scheme, output, exec_ctx)?;
-            trace_ctx.meet_call_end(prev_result);
+        Executed(value) => {
+            let resulted_value = populate_context_from_data(
+                value.clone(),
```

Comment:
This clone is cheap, but one can `match met_result.result` and avoid it.

Reply:
Good catch, thanks